src/main/java/com/example/petbuddybackend/controller/NotificationController.java

```diff
@@ -55,10 +72,22 @@ public Page<SimplyNotificationDTO> getUnreadNotifications(
     @PreAuthorize("isAuthenticated()")
     public SimplyNotificationDTO markNotificationAsRead(@PathVariable Long notificationId,
                                                         @RoleParameter
-                                                  @RequestHeader(value = "${header-name.role}") Role role,
+                                                        @RequestHeader(value = "${header-name.role}") Role role,
                                                         @TimeZoneParameter
-                                                  @RequestHeader(value = "${header-name.timezone}") String timezone) {
+                                                        @RequestHeader(value = "${header-name.timezone}", required = false) String timezone) {
         return notificationService.markNotificationAsRead(notificationId, role, TimeUtils.getOrSystemDefault(timezone));
     }
 
+    @Operation(
+            summary = "Mark all notification of currently logged in user as read",
+            description = "Only notifications for current profile will be marked as read."
+    )
+    @PostMapping("/all-read")
```

Comment:
```suggestion
    @PostMapping("/mark-read")
```
Może podobnie jak dałeś w nazwie funkcji? Imo bardziej czytelne